devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java

```diff
@@ -766,28 +774,33 @@ private void addProject(MavenDevModeLauncher.Builder builder, ResolvedDependency
         }
 
         final Path sourceParent;
-        if (sources.getSourceDirs() == null) {
-            if (sources.getResourceDirs() == null) {
-                throw new MojoExecutionException("The project does not appear to contain any sources or resources");
+        if (sourcePaths.isEmpty()) {
+            if (sources == null || sources.getResourceDirs() == null) {
+                throw new MojoExecutionException(
+                        "Local dependency " + module.toCompactCoords() + " does not appear to have any sources");
```

Comment:
Isn't this a little strict in case of a module that e.g. just unpacks and aggregates some dependencies?

Reply:
Thanks for highlighting this @famod This code is looking weird. Probably a leftover from previous iterations.
I just tried a project with a module w/o sources and it didn't even fail at this point but later complained it hasn't been compiled. I'll open another PR to clean this up and ask for your review.